app/src/main/java/de/danoeh/antennapod/activity/OnlineFeedViewActivity.java

```diff
@@ -150,16 +154,22 @@ protected void onCreate(Bundle savedInstanceState) {
             if (actionBar != null) {
                 actionBar.setTitle(R.string.add_feed_label);
             }
-        } else {
-            throw new IllegalArgumentException("Activity must be started with feedurl argument!");
         }
 
-        Log.d(TAG, "Activity was started with url " + feedUrl);
-        setLoadingLayout();
-        if (savedInstanceState == null) {
-            startFeedDownload(feedUrl, null, null);
+        if (feedUrl == null) {
+            //Toast.makeText(this.getApplicationContext(), "Error: No feed url", Toast.LENGTH_SHORT).show();
+            Log.e(TAG, "feedUrl is null.");
+            NullValueErrorDialogCreator.
+                    newRequestErrorDialog(this,
+                            "feed url is null.");
```

Comment:
I don't think we need a new class for that specific dialog. Creating the dialog here should be enough. After moving, please also delete the now unused Toast comment.